### PR TITLE
chore: Created issue & PR templates

### PR DESCRIPTION
**Pull request by FenrirUnbound**
## Context

Having templates for PRs helps maintain document the intentional changes that a PR introduces. This helps a lot with debugging features in the future, and for those who did not author the feature to determine a good remediation to related issues.

Issue templates help us by asking for the (usually missing) information that allows us to investigate issues more effeciently.

## Objective

Create an initial version of templates for both issues & pull requests.

## References

* https://github.com/blog/2111-issue-and-pull-request-templates

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## Context
+
+<!-- Why do we need this PR? What was the reason that lead you to make this change? -->
```
Comment (r3rastogi) [inline .github/PULL_REQUEST_TEMPLATE.md]: "that led you to"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+## Objective
+
+<!-- What does this PR fix? What intentional changes will this PR make? -->
+
+## References
+
+<!-- Links or resources that help clarify and support your intentions -->
```
Comment (minz1027) [inline .github/PULL_REQUEST_TEMPLATE.md]: So all of us will need to follow this template from now on?

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+** Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
+
+<!--
```
Comment (r3rastogi) [inline .github/ISSUE_TEMPLATE.md]: None of this text shows up in the md file

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (tkyi) [inline .github/PULL_REQUEST_TEMPLATE.md]: Link to the issue you're addressing would be helpful

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (FenrirUnbound) [inline .github/PULL_REQUEST_TEMPLATE.md]: And that's what I get for copy-pasta

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (FenrirUnbound) [inline .github/PULL_REQUEST_TEMPLATE.md]: The templates only pre-populate the information in the description.

It doesn't enforce it 😇

File: .github/ISSUE_TEMPLATE.md (hunk 3 above)
Comment (FenrirUnbound) [inline .github/ISSUE_TEMPLATE.md]: That's why it's a comment. It's meant to give clarification to the person that's opening the issue without adding additional noise to the issue itself.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
+
+<!--
```
Comment (r3rastogi) [inline .github/ISSUE_TEMPLATE.md]: This text still doesn't show up

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+## Context
+
+<!-- Why do we need this PR? What was the reason that led you to make this change? -->
```
Comment (r3rastogi) [inline .github/PULL_REQUEST_TEMPLATE.md]: Same here. None of these descriptions show up in Github.

File: .github/ISSUE_TEMPLATE.md (hunk 4 above)
Comment (FenrirUnbound) [inline .github/ISSUE_TEMPLATE.md]: I mentioned this in your initial comment

> That's why it's a comment. It's meant to give clarification to the person that's opening the issue without adding additional noise to the issue itself.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (tkyi) [inline .github/PULL_REQUEST_TEMPLATE.md]: I guess you could put it in italics and have it show up instead of it being a comment?
🤷‍♀️

Comment (FenrirUnbound) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think it's a lot of extra noise that we don't need to read but the creator should be aware of. Just like comments in code, these comments further clarify what each section represents

Comment (tkyi) [inline .github/PULL_REQUEST_TEMPLATE.md]: But who's going to look at the raw code? 😜

Comment (r3rastogi) [inline .github/PULL_REQUEST_TEMPLATE.md]: But comments in the code are visible haha. I like @tkyi 's idea of putting it in italics and still having it be visible.

Comment (FenrirUnbound) [inline .github/PULL_REQUEST_TEMPLATE.md]: > But who's going to look at the raw code? 

Exactly. 

The idea is to set a mutual expectation that we have the right information in the beginning. 

If you prefer the extra noise, that's trivial to change. Just note that you're the only one who wants it.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (minz1027) [inline .github/PULL_REQUEST_TEMPLATE.md]: 👍

**Pull request merged**